### DISCUSSION TITLE
Extracting a form object is having validation problem

### DISCUSSION THREAD

**Discussion by sayedsadat344** [livewire/livewire#6386] [1. Help] [answered]
I have a component as `AddProduct `where I have extracted a `form object` as mentioned in `livewire-v3` documentation.

`php artisan livewire:form ProductForm`

The ProductForm contains

```
class ProductForm extends Form
{

    #[Rule('required|unique:products|min:5')]
    public $name = '';

    #[Rule('required|min:10')]
    public $description = '';

}
```

AddProduct component contains

```
class AddProduct extends Component
{

public ProductForm $productForm;

   public function saveProduct()
    {

        $this->validate();

        Product::created($this->productForm->all());

        $this->dispatch('alert', [
            'type' => 'success',
            'message' => "Product added Successfully!!",
        ]);

        return $this->redirect('/products');
    }
}
```

The ` required, min validations` are working fine but `unique:products` validation gives an error which is related to the `ProductForm Object.`

Let see the error

```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'productForm.name' in 'where clause'
select count(*) as aggregate from `products` where `productForm`.`name` = Some name
```

What really is happening?

well,` livewire checks required validation for name` which passes the validation, then it checks the` products table` to validate the `uniqueness` that is where the problem arises.

as we can see in the query `select count(*) as aggregate from products where productForm.name = Some name` livewire sends the Object Name (`productForm`) with the property `name (name) concatenated (productForm.name)` as a field in `products table`, which in turn gives an error of `Unknown column 'productForm.name'`.

How can we solve this issue?

**Comment by N1ebieski** [comment, marked as answer]
What about ```unique:products,name``` ?

**Reply by sayedsadat344** [reply]
Thanks for the answer.

It worked with throwing error `The product form.name has already been taken.`

**Comment by peroxovy** [comment]
#6280 problem is quite similar
